### PR TITLE
add book04.word2vec train test

### LINKED ISSUE #5003: need to implement book04 word2vec in our new framework

### PR DESCRIPTION
**Pull request by QiJune**
FIx #5003

### CODE REVIEW

File: python/paddle/v2/framework/tests/test_layers.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def test_fit_a_line(self):
         avg_cost = layers.mean(x=cost, program=program)
         self.assertIsNotNone(avg_cost)
         program.append_backward(avg_cost)
-        print str(program)
+        # print str(program)
```
Comment (reyoung) [inline python/paddle/v2/framework/tests/test_layers.py]: Do not change the previous unit tests.

You can select the test method in runtime, by

```bash
python -m unittest test_layers.TestBook.test_fit_a_line
```

Comment (QiJune) [reply python/paddle/v2/framework/tests/test_layers.py]: Done

File: paddle/framework/var_desc.cc [hunk 2]
```diff
@@ -18,6 +18,10 @@ limitations under the License. */
 namespace paddle {
 namespace framework {
 
+VarDesc::VarType VarDescBind::GetType() const { return desc_.type(); }
```
Comment (wangkuiyi) [inline paddle/framework/var_desc.cc]: I see that we added methods `GetType`  and `SetType`, but I don't see them being used anyway in this PR. Just a kindly reminder -- do we really need them?

Comment (QiJune) [reply paddle/framework/var_desc.cc]: I just modify the syntax of these two methods.
Before, the methods implementation are also in header file. I move the implementation to source file to take consistency with other methods in class VarDescBind.

**Pull request merged**